src/View/AbstractDirectoryView.vala

```diff
@@ -975,34 +971,25 @@ namespace FM {
         }
 
 /** File operation callbacks */
-        static void create_file_done (GLib.File? new_file, void* data) {
-            var view = data as FM.AbstractDirectoryView;
-
+        [CCode (instance_pos = -1)]
+        public void create_file_done (GLib.File? new_file) {
             if (new_file == null) {
                 return;
             }
 
-            if (view == null) {
-                warning ("View invalid after creating file");
-                return;
-            }
             /* Start to rename the file once we get signal that it has been added to model */
-            view.slot.directory.file_added.connect_after (view.after_new_file_added);
-            view.unblock_directory_monitor ();
+            slot.directory.file_added.connect_after (after_new_file_added);
+            unblock_directory_monitor ();
         }
 
-        /** Must pass a pointer to an instance of FM.AbstractDirectoryView as 3rd parameter when
+        /** Must pass a pointer to an instance of FM.AbstractDirectoryView as 2nd parameter when
```

Comment:
Not relevant now?

Reply:
You're right, the annotation is here to inform any C user anyway, I removed the comment.